### PR TITLE
Add warning messages for cluster commands on partitioned tables

### PR DESCRIPTION
**Pull request by hanefi**
PG15 introduces `CLUSTER` commands for partitioned tables. Similar to a `CLUSTER` command with no supplied table names, these commands also can not be run inside transaction blocks and therefore can not be propagated in a distributed transaction block.

I print a hint to advise the users to run the `CLUSTER` command on the partitions. However running commands on partitions are prohibited when replication factor is greater than 1. On such clusters there is no easy way to run the CLUSTER command at all right now.

Relevant PG commit: [cfdd03f45e6afc632fbe70519250ec19167d6765](https://github.com/postgres/postgres/commit/cfdd03f45e6afc632fbe70519250ec19167d6765)

### CODE REVIEW

File: src/backend/distributed/commands/cluster.c [hunk 1]
```diff
@@ -69,6 +70,20 @@ PreprocessClusterStmt(Node *node, const char *clusterCommand,
 		return NIL;
 	}
 
+	/* we do not support CLUSTER command on partitioned tables as it can not be run inside transaction blocks */
+	if (PartitionedTableNoLock(relationId))
```
Comment (marcocitus) [inline src/backend/distributed/commands/cluster.c:74]: maybe we can set `task->cannotBeExecutedInTransction = false;` on the tasks

Comment (hanefi) [reply src/backend/distributed/commands/cluster.c:74]: This was a good starting step, but is not enough. Apparently we can not use `worker_apply_shard_ddl_command` to run CLUSTER commands on shards.

I think we should not spend more time to add support for this, and continue with warning messages.

```
NOTICE:  00000: issuing SELECT worker_apply_shard_ddl_command (102045, 'public', 'CLUSTER sale USING sale_pk;')
DETAIL:  on server hanefi@localhost:9702 connectionId: 2
LOCATION:  LogRemoteCommand, remote_commands.c:351
WARNING:  01000: CONCURRENTLY-enabled index commands can fail partially, leaving behind an INVALID index.
 Use DROP INDEX CONCURRENTLY IF EXISTS to remove the invalid index.
LOCATION:  ExecuteDistributedDDLJob, utility_hook.c:1382
ERROR:  25001: CLUSTER cannot be executed from a function
CONTEXT:  while executing command on localhost:9702
LOCATION:  ReportResultError, remote_commands.c:318
```

Comment (onderkalaci) [reply src/backend/distributed/commands/cluster.c:74]: I think we can still do that, we should also avoid `worker_apply_shard_ddl_command` such as we do for `VACUUM` -- `VacuumTaskList()`.

I'm fine with skipping it today given the code-freeze.
 
maybe clarify the comment further:

```
PostgreSQL currently does not support CLUSTER command on partitioned tables in a transaction block. Although Citus can execute commands outside of transaction block -- such as VACUUM -- we cannot do that here because CLUSTER command is also not allowed from a function call as well. By default, Citus uses `worker_apply_shard_ddl_command()`, where we should avoid it for this case.
```

Comment (onderkalaci) [inline src/backend/distributed/commands/cluster.c:74]: why do you prefer `PartitionedTableNoLock` over `PartitionedTable`? We seem to use the former for performance critical code-paths, which seems irrelevant here.

**Review by onderkalaci — APPROVED** [review]

File: src/backend/distributed/commands/cluster.c (hunk 1 above)
Comment (hanefi) [reply src/backend/distributed/commands/cluster.c:74]: Several lines earlier, we already got an AccessExclusive lock. I assumed we no longer need to acquire another lock. I believe it is highly unlikely that we stop getting that lock there in the future.

https://github.com/citusdata/citus/blob/b79111527e70fff309f9df3c9067620168bc7605/src/backend/distributed/commands/cluster.c#L52-L54
